### PR TITLE
feat(select): search locale support for  #830

### PR DESCRIPTION
**Pull request by cenarturkmen**
I added toLocaleLowerCase method for the search function of select. 
Also created a util method for getting locale. I'm not sure why we are not exporting getLocale from "configureLocalization"

### CODE REVIEW

**Comment by cenarturkmen** [comment]
I couldn't link but issue: [830](https://github.com/Trendyol/baklava/issues/830)

File: src/localization.ts [hunk 1]
```diff
@@ -14,11 +15,9 @@ const { setLocale } = configureLocalization({
 
 export const init = async () => {
   const html = document.querySelector("html");
```
Comment (erbilnas) [inline src/localization.ts:17]: I believe we can do away with this line now

File: src/localization.ts [hunk 2]
```diff
@@ -1,6 +1,7 @@
 import { LocaleModule, configureLocalization } from "@lit/localize";
 import { sourceLocale, targetLocales, allLocales } from "./generated/locale-codes";
 import * as templatesTR from "./generated/locales/tr";
+import { getLocale } from "./utilities/getLocale";
 
 type LangKey = "tr" | "en";
```
Comment (erbilnas) [inline src/localization.ts:6]: this one too

**Review by erbilnas — CHANGES_REQUESTED** [review]

File: src/utilities/getLocale.ts [hunk 3]
```diff
@@ -0,0 +1,14 @@
+import { allLocales } from "../generated/locale-codes";
+
+type LangKey = "tr" | "en";
+
+export const getLocale = (): LangKey => {
```
Comment (ogunb) [inline src/utilities/getLocale.ts:5]: `configureLocalization` exposes a getLocale method. We can export that from localization.ts and use that instead. Issue detail might have been misleading sorry about that.

**Review by ogunb — CHANGES_REQUESTED** [review]

**Comment by ogunb** [comment]
Also, would you be able to document this behavior in Storybook?

Thanks for your contribution btw

**Review by ozkersemih — APPROVED** [review]

**Comment by buseselvi** [comment]
Closed this as completed in [#857](https://github.com/Trendyol/baklava/pull/857). If there is something missing, we can open an issue separately. Thank you 🙏 @cenarturkmen

**Pull request closed without merging**